cli/i18n/I18nData.php

```diff
@@ -14,20 +14,25 @@ public function __construct(array $data) {
 		$this->processValueStates();
 	}
 
-	public function getData() {
+	/**
+	 * @return array<string|array<string>|array<array<string>>>
+	 */
+	public function getData(): array {
 		return $this->data;
 	}
 
-	private function addMissingKeysFromReference() {
+	private function addMissingKeysFromReference(): void {
 		$reference = $this->getReferenceLanguage();
 		$languages = $this->getNonReferenceLanguages();
 
 		foreach ($reference as $file => $refValues) {
+			/** @var I18nValue $refValue */
```

Comment:
It should not be necessary to manually override the types there. It is likely a sign of problem for the signature of `$this->getReferenceLanguage()`